whylog_vim/proxy/log_reader/__init__.py

```diff
@@ -1,24 +1,41 @@
+import six
+from whylog.log_reader.exceptions import NoLogTypeError
+from whylog_vim.consts import EditorStates, Messages
 from whylog_vim.output_formater.log_reader_formater import LogReaderOutput
 
 
 class LogReaderProxy(object):
-    def __init__(self, editor, log_reader):
-        self.editor = editor
+    def __init__(self, log_reader, config, editor, main_proxy):
         self._log_reader = log_reader
+        self._config = config
+        self._editor = editor
+        self._main_proxy = main_proxy
+
+    def _ask_log_type(self, log_types):
+        pass
 
     def new_query(self):
-        front_input = self.editor.get_front_input()
-        query_output = self._log_reader.get_causes(front_input)
-        self.output = LogReaderOutput.format_query(front_input, query_output)
-        self.output.set_default_callback_function(self.editor.close_query_window)
-        self.editor.create_query_window(self.output.get_content())
-        self.editor.go_to_query_window()
+        front_input = self._editor.get_front_input()
+        try:
+            query_output = self._log_reader.get_causes(front_input)
+        except NoLogTypeError:
+            log_types = [logType for logType in self._config.get_all_log_types()]
+            if log_types:
+                self._ask_log_type(log_types)
+            else:
```

Comment:
Right. It is supposed to work like this:
when an unrecognized log is opened and F3/F4 function is used, a prompt should appear and ask the user whether he wants to select one of the existing log types for that log or define a new type of log. There is a "default" type of log to lower the entry barreir for the user.

User might then:
- choose "default"
- choose one of the existing log types temporarily
- choose one of the existing log types and modify the log name pattern, so that next time this file is recognized properly (so it's an "edit log name pattern" function)
- choose to create a new log type and
  - define a pattern for it
  - choose it temporarily